components/scripts/gradle/01-validate-incremental-building.sh

```diff
@@ -106,23 +121,57 @@ wizard_execute() {
   print_bl
   explain_measure_build_results
   print_bl
+  fetch_build_cache_metrics
+  print_bl
   explain_and_print_summary
 }
 
+# shellcheck disable=SC2086 # splitting expected
 execute_first_build() {
   info "Running first build:"
-  info "./gradlew --no-build-cache --scan -Dscan.tag.${EXP_SCAN_TAG} -Dscan.value.runId=${RUN_ID} clean ${tasks}$(print_extra_args)"
-
-  # shellcheck disable=SC2086  # we want tasks to expand with word splitting in this case
-  invoke_gradle --no-build-cache clean ${tasks}
+  execute_build clean ${tasks}
 }
 
+# shellcheck disable=SC2086 # splitting expected
 execute_second_build() {
   info "Running second build:"
-  info "./gradlew --no-build-cache --scan -Dscan.tag.${EXP_SCAN_TAG} -Dscan.value.runId=${RUN_ID} ${tasks}$(print_extra_args)"
+  execute_build ${tasks}
+}
+
+execute_build() {
+  print_gradle_command "$@"
+  invoke_gradle --no-build-cache "$@"
+}
 
-  # shellcheck disable=SC2086  # we want tasks to expand with word splitting in this case
-  invoke_gradle --no-build-cache ${tasks}
+print_gradle_command() {
+  local scan_arg
+  if [[ "${build_scan_publishing_mode}" == "on" ]]; then
+    scan_arg="--scan"
+  else
+    scan_arg="-Dscan.dump"
+  fi
+  info "./gradlew --no-build-cache ${scan_arg} -Dscan.tag.${EXP_SCAN_TAG} -Dscan.value.runId=${RUN_ID} $*$(print_extra_args)"
+}
+
+fetch_build_cache_metrics() {
+  if [ "$build_scan_publishing_mode" == "on" ]; then
+    read_build_scan_metadata
+    fetch_and_read_build_scan_data build_cache_metrics_only "${build_scan_urls[@]}"
+  else
+    find_and_read_build_scan_dumps
+  fi
+}
+
+# Overrides info.sh#print_performance_metrics
+print_performance_metrics() {
+  print_performance_characteristics
+}
+
+# Overrides info.sh#print_performance_characteristics
+print_performance_characteristics() {
```

Comment:
0: I considered suggesting to inline this function in the overridden `print_performance_metrics` function above, but doing it this way ensures the right thing happens if ``print_performance_characteristics` is called elsewhere (now, or in the future).

(This comment is for the benefit of other reviewers).

Reply:
I actually decided to remove the `print_performance_metrics` function in favor of `print_performance_characteristics` in a subsequent PR (#283) as it was ultimately redundant.